kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/storage/cloud/KieServerStateOpenShiftRepository.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.openshift.impl.storage.cloud;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.kie.server.services.impl.storage.KieServerState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KieServerStateOpenShiftRepository extends KieServerStateCloudRepository {
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServerStateOpenShiftRepository.class);
+
+    public synchronized void create(@NotNull KieServerState kieServerState) {
+        String serverId = retrieveKieServerId(kieServerState);
+        processKieServerState(client -> {
+            ConfigMap cm = client.configMaps().withName(serverId).get();
+
+            String stateXML = xs.toXML(kieServerState);
+            if (cm != null) {
+                logger.info("Create new ConfigMap action ignored. ConfigMap for KieServer [{}] exists.", serverId);
+                return null;
+            }
+
+            client.configMaps().create(new ConfigMapBuilder()
+                                                             .withNewMetadata()
+                                                             .withName(serverId)
+                                                             .endMetadata()
+                                                             .withData(Collections.singletonMap(CFG_MAP_DATA_KEY, stateXML))
+                                                             .build());
+            return null;
+        });
+    }
+
+    public List<String> retrieveAllKieServerIds() {
+        return processKieServerState(client -> {
+            return client.configMaps().list().getItems()
+                         .stream()
+                         .map(cfg -> cfg.getMetadata().getName())
```

Comment:
Same here, the code doesn't count with the case when config map is used for other things than Kie server state.